recursion/recurse.rb

```diff
@@ -1,10 +1,32 @@
 def fact(n)
+  # base case: 
+  if n == 0 || n == 1
+    return 1
+  end
+    return n * fact(n - 1)
 end
 
 def fib(n)
+  # base cases:
+  if n == 0
+    return 0
```

Comment:
Hmm... this is interesting. I would say that you can't call fib(0) so this case isn't necessary, as it should be an assumption than n > 0